kafka/hash.go

```diff
@@ -1,12 +1,13 @@
 package kafka
 
-import kafkalib "github.com/confluentinc/confluent-kafka-go/kafka"
-
-func GetPartition(msg *kafkalib.Message, partitions ...int) (partition int) {
+func GetPartition(key string, partitions []int32) int32 {
+	if len(partitions) == 0 {
+		return -1
+	}
 	// NOTE: the murmur2 balancers in java and librdkafka treat a nil key as
 	//       non-existent while treating an empty slice as a defined value.
-	idx := (murmur2(msg.Key) & 0x7fffffff) % uint32(len(partitions))
-	return partitions[idx]
+	idx := (murmur2([]byte(key)) & 0x7fffffff) % uint32(len(partitions))
```

Comment:
wouldn't you need to use the configured algorithm here?

Reply:
We will need to add additional hashing algorithms to support reading from a partition.  So you made me think we should track this and throw an error if the hashing balancer is not set to murmur2